quinn-h3/benches/bench.rs

```diff
@@ -119,54 +119,8 @@ impl Context {
         server.listen(addr);
         debug!("server bind");
         let handle = thread::spawn(move || {
-            let my_span = span!(Level::TRACE, "server");
-            let _enter = my_span.enter();
             let mut runtime = rt();
-            let handle = runtime.spawn(
-                async move {
-                    let mut incoming = server.build().unwrap();
-                    let mut incoming_req = incoming
-                        .next()
-                        .await
-                        .expect("accept")
-                        .await
-                        .expect("connect");
-                    debug!("recv incoming");
-                    while let Some(recv_req) = incoming_req.next().await {
-                        let (request, mut body_reader, sender) = recv_req.await.expect("recv_req");
-                        while let Some(_) = body_reader.data().await {}
-
-                        let mut body_writer = sender
-                            .send_response(
-                                Response::builder().status(StatusCode::OK).body(()).unwrap(),
-                            )
-                            .await
-                            .expect("send_response");
-
-                        let frame_size = request
-                            .headers()
-                            .get("frame_size")
-                            .map(|x| x.to_str().unwrap().parse().expect("parse frame size"))
-                            .expect("no frame size");
-                        let mut remaining = request
-                            .headers()
-                            .get("total_size")
-                            .map(|x| x.to_str().unwrap().parse().expect("parse total size"))
-                            .expect("no total size");
-
-                        let data = "a".repeat(frame_size);
-                        while remaining > 0 {
-                            let size = std::cmp::min(frame_size, remaining);
-                            body_writer
-                                .write_all(&data.as_bytes()[..size])
-                                .await
-                                .expect("body write");
-                            remaining -= size;
-                        }
-                    }
-                }
-                .instrument(error_span!("server")),
-            );
+            let handle = runtime.spawn(handle_connection(server).instrument(error_span!("server")));
```

Comment:
This is what we meant:

```suggestion
            let handle = runtime.block_on(handle_connection(server).instrument(error_span!("server"))).unwrap();
```

Reply:
Yeah, sorry but I got that moved to the first commit.

Reply:
@Ralith Is it ok for you ?

Reply:
Yep, all sorted. Thanks!